### ISSUE TITLE
System.AggregateException when calling function requiring an array of integers as a parameter.

### ISSUE THREAD

**Issue by jamesra** [object/Simple.OData.Client#340]
On my server I have a function with this meta-data signature:

```
<Function Name="NetworkCells">
<Parameter Name="IDs" Type="Collection(Edm.Int64)" Nullable="false"/>
<Parameter Name="Hops" Type="Edm.Int32" Nullable="false"/>
<ReturnType Type="Collection(ConnectomeDataModel.Structure)"/>
</Function>
```

I am attempting to call this function using the V4 client:

```
parameters["Hops"] = 1;
parameters["IDs"] = new long[] { 180 };
var structures = await client.ExecuteFunctionAsArrayAsync<Structure>("NetworkCells", parameters);
```

This throws a System.AggregateException with the message ""The type 'System.Int64[]' is not supported when converting to a URI literal."

The call succeeds when using this URL, or the FindEntriesAsync implementation:

> …/OData/NetworkCells(IDs=@IDs,Hops=1)?@IDs=[476]
>
> var structures = await client.FindEntriesAsync("NetworkCells(IDs=@IDs,Hops=1)?@IDs=[476]");
> Debug.Assert(structures != null);

The workaround appears to be usable.  Not yet sure if I'm doing something unsupported or if this is a bug.

Below is the stack trace from the inner exception:

>    at Microsoft.OData.Core.Evaluation.LiteralFormatter.FormatRawLiteral(Object value)
>    at Microsoft.OData.Core.Evaluation.LiteralFormatter.FormatAndEscapeLiteral(Object value)
>    at Microsoft.OData.Core.Evaluation.LiteralFormatter.DefaultLiteralFormatter.FormatLiteralWithTypePrefix(Object value)
>    at Microsoft.OData.Core.Evaluation.LiteralFormatter.DefaultLiteralFormatter.Format(Object value)
>    at Microsoft.OData.Core.UriParser.ODataUriConversionUtils.ConvertToUriPrimitiveLiteral(Object value, ODataVersion version)
>    at Microsoft.OData.Core.UriParser.ODataUriUtils.ConvertToUriLiteral(Object value, ODataVersion version, IEdmModel model)
>    at Simple.OData.Client.V4.Adapter.CommandFormatter.<>c__DisplayClass3_0.<ConvertValueToUriLiteral>b__0(Object x)
>    at Simple.OData.Client.V4.Adapter.CommandFormatter.ConvertValueToUriLiteral(Object value, Boolean escapeDataString)
>    at Simple.OData.Client.CommandFormatterBase.<ConvertKeyValuesToUriLiteral>b__7_1(KeyValuePair`2 x)
>    at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
>    at System.String.Join(String separator, IEnumerable`1 values)
>    at Simple.OData.Client.CommandFormatterBase.ConvertKeyValuesToUriLiteral(IDictionary`2 key, Boolean skipKeyNameForSingleValue)
>    at Simple.OData.Client.CommandFormatterBase.FormatCommand(FluentCommand command)
>    at Simple.OData.Client.FluentCommand.Format()
>    at Simple.OData.Client.FluentCommand.<GetCommandTextAsync>d__18.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at Simple.OData.Client.ODataClient.<ExecuteFunctionAsync>d__150.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at Simple.OData.Client.ODataClient.<ExecuteFunctionAsEnumerableAsync>d__76.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at Simple.OData.Client.ODataClient.<ExecuteFunctionAsArrayAsync>d__80`1.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
>    at AnnotationVizLib.SimpleODataMorphologyFactory.<FromOData>d__0.MoveNext() in C:\Src\Git\Viking\AnnotationVizLib\Factories\SimpleOData\SimpleODataMorphologyFactory.cs:line 36

Thanks for the great library!

**Comment by jamesra** [comment]
Full disclosure:  I switched to Simple.OData.Client after I was unable to use array parameters with the Micrsoft.OData.Client library.  I believed the underlying causes were unrelated.  As I dug into the code it appears both use Microsoft OData libraries to encode the parameters into the URI.  I do not know if there is a single root cause but I've linked the other issue below just in case: 

https://github.com/OData/odata.net/issues/725
